# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/stdint.h.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- dietlibc-0.31-orig/include/stdint.h	2005-03-15 09:51:22.000000000 +0100
+++ dietlibc-0.31/include/stdint.h	2009-05-01 11:48:33.000000000 +0200
@@ -31,8 +31,23 @@
    should only be defined if explicitly requested.  */
 #if !defined __cplusplus || defined __STDC_LIMIT_MACROS
 
+/* Minimum of signed integral types.  */
+# define INT8_MIN               (-128)
+# define INT16_MIN              (-32767-1)
+# define INT32_MIN              (-2147483647-1)
+
+/* Maximum of signed integral types.  */
+# define INT8_MAX               (127)
+# define INT16_MAX              (32767)
+# define INT32_MAX              (2147483647)
+
+/* Maximum of unsigned integral types.  */
+# define UINT8_MAX              (255)
+# define UINT16_MAX             (65535)
+# define UINT32_MAX             (4294967295U)
+
 /* FIXME: missing are
-     INTx_MIN, INTx_MAX, UINTx_MAX
+     INT64_MIN, INT64_MAX, UINT64_MAX
      INT_LEASTx_MIN, INT_LEASTx_MAX, UINT_LEASTx_MAX
      INT_FASTx_MIN, INT_FASTx_MAX, UINT_FASTx_MAX
      INTMAX_MIN, INTMAX_MAX, UINTMAX_MAX
